react/features/video-layout/subscriber.js

```diff
@@ -2,12 +2,47 @@
 
 import debounce from 'lodash/debounce';
 
+import { getSourceNameSignalingFeatureFlag } from '../base/config';
 import { StateListenerRegistry, equals } from '../base/redux';
 import { isFollowMeActive } from '../follow-me';
 
-import { setRemoteParticipantsWithScreenShare } from './actions';
+import { setRemoteParticipantsWithScreenShare, setRemoteFakeScreenShareParticipants } from './actions';
 import { getAutoPinSetting, updateAutoPinnedParticipant } from './functions';
 
+StateListenerRegistry.register(
+    /* selector */ state => [ ...state['features/base/participants'].sortedFakeScreenShareParticipants.values() ],
+    /* listener */ debounce((sortedFakeScreenShareParticipants, store) => {
+        if (!getAutoPinSetting() || isFollowMeActive(store) || !getSourceNameSignalingFeatureFlag(store.getState())) {
+            return;
+        }
+
+        const oldScreenSharesOrder = store.getState()['features/video-layout'].remoteScreenShares || [];
+        const knownSharingParticipantIds = sortedFakeScreenShareParticipants.map(p => p.id);
+
+        // Filter out any participants which are no longer screen sharing
+        // by looping through the known sharing participants and removing any
+        // participant IDs which are no longer sharing.
+        const newScreenSharesOrder = oldScreenSharesOrder.filter(
+            participantId => knownSharingParticipantIds.includes(participantId));
+
+        // Make sure all new sharing participant get added to the end of the
+        // known screen shares.
+        knownSharingParticipantIds.forEach(participantId => {
+            if (!newScreenSharesOrder.includes(participantId)) {
+                newScreenSharesOrder.push(participantId);
+            }
+        });
+
+        if (!equals(oldScreenSharesOrder, newScreenSharesOrder)) {
+            store.dispatch(setRemoteFakeScreenShareParticipants(newScreenSharesOrder));
```

Comment:
I think we can remove this new action and use the existing `setRemoteParticipantsWithScreenShare` action. We are also only listening to changes for `sortedFakeScreenShareParticipants`, we will need to also auto pinning when `localScreenShare` changes following the skipTrack logic below

Reply:
Can this be supported in a follow up PR?